src/main/java/org/asamk/signal/http/HttpServerHandler.java

```diff
@@ -0,0 +1,123 @@
+
+    private final ObjectMapper objectMapper = Util.createJsonObjectMapper();
+
+    private final InetSocketAddress address;
+
+    private final Manager m;
+
+    private final MultiAccountManager c;
+
+    public HttpServerHandler(final InetSocketAddress address, final Manager m) {
+        this.address = address;
+        this.m = m;
+        this.c = null;
+    }
+
+    public HttpServerHandler(final InetSocketAddress address, final MultiAccountManager c) {
+        this.address = address;
+        this.m = null;
+        this.c = c;
+    }
+
+    public void init() {
+
+        try {
+
+            logger.info("Starting server on port " + address.toString());
+
+            final var server = HttpServer.create(new InetSocketAddress(address.getPort()), 0);
+            server.setExecutor(Executors.newFixedThreadPool(10));
+
+            server.createContext("/api/v1/rpc", httpExchange -> {
+
+                if (!"POST".equals(httpExchange.getRequestMethod())) {
+                    sendResponse(405, null, httpExchange);
+                    return;
+                }
+
+                try {
+
+                    final SignalJsonRpcCommandHandler commandHandler;
+
+                    if (c != null) {
+                        commandHandler = new SignalJsonRpcCommandHandler(c, Commands::getCommand);
+                    } else {
+                        commandHandler = new SignalJsonRpcCommandHandler(m, Commands::getCommand);
+                    }
+
+                    final Object[] result = {null};
+                    final var jsonRpcSender = new JsonRpcSender(s -> {
+                        if (result[0] != null) {
+                            throw new AssertionError("There should only be a single JSON-RPC response");
+                        }
+
+                        result[0] = s;
+                    });
+
+                    final var jsonRpcReader = new JsonRpcReader(jsonRpcSender, httpExchange.getRequestBody());
+                    jsonRpcReader.readMessages((method, params) -> commandHandler.handleRequest(objectMapper, method, params),
+                            response -> logger.debug("Received unexpected response for id {}", response.getId()));
+
+                    if (result[0] !=null) {
+                        sendResponse(200, result[0], httpExchange);
+                    } else {
+                        sendResponse(201, null, httpExchange);
+                    }
+
+                }
+                catch (Throwable aEx) {
+                    logger.error("Failed to process request.", aEx);
+                    sendResponse(200, JsonRpcResponse.forError(
+                            new JsonRpcResponse.Error(JsonRpcResponse.Error.INTERNAL_ERROR,
+                            "An internal server error has occurred.", null), null), httpExchange);
+                }
+            });
+
+            server.start();
+
+        } catch (Throwable ex) {
+            ex.printStackTrace();
+        }
```

Comment:
I don't like this outer try/catch block. If the server fails to start, signal-cli should exit with an error message.
Just make `init()` throws IOException, catch that in the DaemonCommand and rethrow it wrapped in an `IOErrorException`.

Reply:
Yea, makes sense. Did that.